src/append_only_data.rs

```diff
@@ -969,4 +996,64 @@ mod tests {
         let decoded = unwrap!(self::Address::decode_from_zbase32(&encoded));
         assert_eq!(address, decoded);
     }
+
+    #[test]
+    fn append_unseq_data_test() {
+        let mut data = UnpubUnseqAppendOnlyData::new(XorName(rand::random()), 10);
+
+        // Assert that the current Append Operation has duplicate entries present and are not pushed
+        let entries = [
+            (b"KEY1".to_vec(), b"VALUE1".to_vec()),
+            (b"KEY2".to_vec(), b"VALUE2".to_vec()),
+            (b"KEY1".to_vec(), b"VALUE1".to_vec()),
+        ]
+        .to_vec();
+        assert!(data.append(entries).is_err());
```

Comment:
It'd be more thorough to check we're getting the correct variant returned:
```rust
assert_eq!(Error::DuplicateEntryKeys, unwrap_err!(data.append(entries)));
```
Also, it might be worth checking the state of `data.entries()` is as expected?  Even just once at the very end of the test, or after each attempt to mutate the data.